src/js/menuItem.js

```diff
@@ -0,0 +1,98 @@
+import { getMenuItemsData } from './fetch.js';
+
+let menuData = []; // store all menu items globally
+
+async function getMenuData() {
+    menuData = await getMenuItemsData();
+    if (!menuData || menuData.length === 0) {
+        const container = document.querySelector('.menu-items');
+        const errorMsg = document.createElement('p');
+        errorMsg.textContent = 'No dining data available. Please try again later.';
+        errorMsg.style.color = 'red';
+        errorMsg.style.fontSize = '40px';
+        errorMsg.style.textAlign = 'center';
+        container.appendChild(errorMsg);
+        return;
+    }
+
+    populateTagDropdown();
+    renderMenuItems();
+    renderSpecialItems();
+}
+
+function populateTagDropdown() {
+    const tagSelect = document.getElementById('tagFilter');
```

Comment:
```suggestion
    const tagSelect = document.querySelector('#tagFilter');
```
Op deze manier houd je het consistent met de functie hiervoor

Reply:
Yes, heb het nu aangepast!